synapse/models/infotech.py

```diff
@@ -1,79 +1,373 @@
+                        ('url', {'ptype': 'inet:url'}),
+                    ]),
+
+                    ('it:av:filehit', {'ptype': 'it:av:filehit'}, [
+                        ('file', {'ptype': 'file:bytes'}),
+                        ('sig', {'ptype': 'it:av:sig'}),
+                    ])
+                ),
+            }),
+
+            ('it', 201708161120, {
+
+                'types':(
+
+                    ('it:dev:str', {'subof':'str', 'doc':'A developer selected string'}),
+                    ('it:dev:int', {'subof':'int', 'doc':'A developer selected int constant'}),
+
+                    ('it:exec:proc', {'subof':'guid', 'doc':'A unique process execution on a host'}),
+
+                    ('it:dev:pipe', {'subof':'it:dev:str', 'doc':'A  named pipe string'}),
+                    ('it:dev:mutex', {'subof':'it:dev:str', 'doc':'A named mutex string'}),
+                    ('it:dev:regkey', {'subof':'it:dev:str', 'doc':'A windows registry key string'}),
+
+                    ('it:dev:regval', {'subof':'comp', 'fields':'key=it:dev:regkey',
+                                       'optfields':'str=it:dev:str,int=int,bytes=file:bytes',
+                                       'doc':'A windows registry key/val pair'}),
+
+                    #('it:exec:file', {'subof':'comp', 'fields':'host=it:host',
+                                      #'optfields':'path=file:path,file=file:bytes,time=time,proc=it:exec:proc',
+                                      #'doc':'A file instance created on a host'}),
+
+                    ('it:exec:file:create', {'subof':'comp', 'fields':'host=it:host',
+                                      'optfields':'path=file:path,file=file:bytes,time=time,exe=file:bytes,proc=it:exec:proc',
+                                      'doc':'A file create event on a host'}),
+
+                    ('it:exec:file:delete', {'subof':'comp', 'fields':'host=it:host',
+                                      'optfields':'path=file:path,file=file:bytes,time=time,exe=file:bytes,proc=it:exec:proc',
+                                      'doc':'A file delete event on a host'}),
+
+                    ('it:exec:file:read', {'subof':'comp', 'fields':'host=it:host',
+                                      'optfields':'path=file:path,file=file:bytes,time=time,exe=file:bytes,proc=it:exec:proc',
+                                      'doc':'A file read event on a host'}),
+
+                    ('it:exec:file:write', {'subof':'comp', 'fields':'host=it:host',
+                                      'optfields':'path=file:path,file=file:bytes,time=time,exe=file:bytes,proc=it:exec:proc',
+                                      'doc':'A file write event on a host'}),
+
+                    ('it:exec:pipe', {'subof':'comp', 'fields':'host=it:host,name=it:dev:pipe',
+                                       'optfields':'exe=file:bytes,proc=it:exec:proc,time=time',
+                                      'doc':'A named pipe instance created on a host'}),
```

Comment:
i'd like to keep the distinction since there are questions that are more easily answered by specific types ( as long as the strings are folding together into it:dev:str nodes )